packages/pie-toolbox/src/code/editable-html/editor.jsx

```diff
@@ -420,18 +421,74 @@ export class Editor extends React.Component {
   };
 
   onEditingDone = () => {
-    if (this.state.isHtmlMode) {
+    const { isHtmlMode, dialog, value, pendingImages } = this.state;
+
+    // Handling HTML mode and dialog state
+    if (isHtmlMode && !dialog?.open) {
+      const currentValue = htmlToValue(value.document.text);
+      const previewText = this.renderPreviewText(currentValue.document.text);
+
+      this.openDialogWithPreview(currentValue, previewText);
       return;
     }
 
-    const { pendingImages } = this.state;
+    // Early return if HTML mode is enabled
+    if (isHtmlMode) return;
 
+    // Handling pending images
     if (pendingImages.length) {
-      this.setState({ scheduled: true });
+      this.scheduleImageProcessing();
```

Comment:
I like the fact that you explicitly named what this setState is supposed to do, but scheduleImageProcessing only does a set state. I think it would be easier to read if we keep it as it was, but maybe add a comment 
// schedule image processing